src/main/java/io/stargate/sgv2/jsonapi/service/projection/DocumentProjector.java

```diff
@@ -191,36 +230,36 @@ static PathCollector collectPaths(JsonNode def, boolean includeSimilarity) {
     }
 
     public DocumentProjector buildProjector() {
-      if (isDefaultProjection()) {
-        return defaultProjector();
-      }
-
       // One more thing: do we need to add document id?
       if (inclusions > 0) { // inclusion-based projection
-        // doc-id included unless explicitly excluded
         return new DocumentProjector(
-            ProjectionLayer.buildLayersNoOverlap(paths, slices, !Boolean.FALSE.equals(idInclusion)),
+            ProjectionLayer.buildLayersForProjection(
+                paths,
+                slices,
+                // doc-id included unless explicitly excluded
+                !Boolean.FALSE.equals(idInclusion),
```

Comment:
This is interesting.
So by default, idInclusion will be null, which will result as **addDocId** as
- true for inclusion-based projection
- false for exclusion-based projection

So essentially the meaning of **addDocId** will be flipped, in terms of what projection is used `inclusion-based` or `exclusion-based`. The variable name addDocId (similarly add$vector, add$vectorize) gives some confusion to me initially, but it makes sense afterwards.

Reply:
Good point, naming is hard. Instead of "addDocId" could use "includeDocId" but that would be even more confusing.
If you (or anyone else) has better name suggestion would be happy to rename.